wolfcrypt/src/asn.c

```diff
@@ -6122,6 +6122,14 @@ static WC_INLINE int GetTime(int* value, const byte* date, int* idx)
 int ExtractDate(const unsigned char* date, unsigned char format,
                                                   struct tm* certTime, int* idx)
 {
+    /* Extract the time from the struct tm - 16bit processors store as uint8_t */
+    int tm_year = certTime->tm_year;
```

Comment:
Why add 6 new variables on the stack, copy into them, use them, and then write them out?  That seems highly inefficient?

Reply:
The avr-lib using 8bit integers for these values (except for year, just there for completeness), and the functions below this are expecting "int" values which are 16bit integers values on a 16bit microcontroller, so they must be converted the re-populated back.